### DISCUSSION TITLE
Question How to use IPolygon Contains IPoint NetTopologySuite With MSSQL EntityFrameworkCore2.1

### DISCUSSION THREAD

**Discussion by tonproperdi** [NetTopologySuite/NetTopologySuite#557] [Q&A]
I try to query address is have location in side polygon point by this code

 ```C#
var geometryFactory = NtsGeometryServices.Instance.CreateGeometryFactory(srid: 4326);

//locations = [ { Lat:72.5, Long:14.2} , { Lat:72.5 , Long:14.8}, { Lat:72.8, Long:14.2} ,{ Lat:72.5, Long:14.2}] 

var coordinates = locations.Select(s => new Coordinate { X = s.Long, Y = s.Lat , Z = 0  }).ToArray();

var LLong = 14.3;

var LLat = 72.6; 

Point pPoint = new Point(x : LLong, y: LLat); 

var polygonLocation = geometryFactory.CreatePolygon(coordinates); 

var isContain =  polygonLocation.Contains(pPoint);//Work return true

Address = Address.Where(w => polygonLocation.Contains(w.Location)); //EF Core Doesn't work

//* w.Location  type is Point
```

If not use in ef linq query it work fine have no error. If use with ef linq query it have this error message.

> System.Data.SqlClient.SqlError: Cannot call methods on varbinary.

### System Configuration

SSMS V18 preview5 
.NetCore 2.1 
NetTopologySuite 2.2.4
DevOS Windows10
VSO2019 Community

**Comment by woodsyb-zz** [comment]
This has nothing to do with NTS really. Or linq. This is an issue with the sql geom type I think.
What is defintion of Address? A Point object in sql server != a Point object in NTS. You will need to do sqlPoint.STAsBinary() and then create a NTS geom object with a WKBReader in order to convert the Sql Objects to NTS ones. Then you can call things like Contains.....

The flip side is if you use a SqlServer STGeomFromText to make the Polygon then you could call
sqlPoly.STContains(sqlPoint) and not use NTS at all.

**Comment by tonproperdi** [comment]
Ok then in sql use like this 


```
DECLARE @g geography;  
SET @g = geography::STPolyFromText ('POLYGON(( -122 47 , -122  46 , -121  46 ,  -122 47))', 4326);  
 
DECLARE @LPoint AS geography = geography::STGeomFromText('POINT(-121.1 46.01 )',4326)
SELECT  @g ,@LPoint  
SELECT    @g.STContains(@LPoint)
```

right?

**Comment by woodsyb-zz** [comment]
Yes. You also may want to consider making them a geometry object rather than geography. Things like areas and distances are much more accurate in Geography but stuff like intersections will be much faster with Geometry objects and will be just as accurate. 

And if you can have tables with this stuff you can do things like 
`select * from PointsTable where @g.STContains(geom)`
or if they are both in tables you can even do stuff like 
`select a.*,b.* from PointsTable a join PolyTable b on b.geom.STContains(a.geom)`

But if you are doing stuff like that you would want to make sure you build a spatial index on the PolyTable.

**Comment by Kum4r4n** [comment]
@HornedSalmon 
@tonproperdi 
Can anyone explain this to me? 
I had the same problem, I want to find Is Point Within a polygon   How can I Find It?
Please give me a complete solution!
